taxi/views.py

```diff
@@ -79,6 +80,17 @@
     success_url = reverse_lazy("taxi:car-list")
 
 
+class AddOrDeleteUser(LoginRequiredMixin, View):
+    def post(self, request, pk):
+        car = get_object_or_404(Car, pk=pk)
+        if car.drivers.filter(id=request.user.id).exists():
+            car.drivers.remove(request.user)
+        else:
+            car.drivers.add(request.user)
+
+        return redirect("taxi:car-detail", pk=pk)
```

Comment:
The `AddOrDeleteUser` view toggles the association of the current user with a car. Ensure that the logic correctly handles edge cases, such as when a user is not allowed to be added or removed from a car due to business rules.